### DISCUSSION TITLE
GraphQL query for MatrixFields

### DISCUSSION THREAD

**Discussion by DTesch-Level-R** [craftcms/cms#11511] [Q&A] [answered]
Is it possible to query for matrixfields using GraphQL?
I want to filter the results based on the matrixfield value.

**Comment by brandonkelly** [comment]
It’s not. You should be able to accomplish what you want using the [Preparse Field](https://plugins.craftcms.com/preparse-field) plugin, which can be used to store prerendered values as top-level fields. So for example if you want to filter entries in GraphQL if any of its Matrix blocks contain a certain value, you could set the Preparse field to:

```twig
{{ clone(element.myMatrixField)
  .someField('test-value')
  .exists() ? '1' }}
```

Then in your GraphQL query, you could include `myPreparseField: 1` in the params.

**Comment by DTesch-Level-R** [comment]
Thanks but that won't work for me because i am using the CMS in headless mode.

**Reply by brandonkelly** [reply]
My suggestion will work either way.

**Reply by DTesch-Level-R** [reply]
Is it possible to make it function like a search? The user should be able to search for any string values in the front end.

**Reply by brandonkelly** [reply]
Sorry, if you just want this for `search`, Preparse won’t be needed. Just posted a [new answer](https://github.com/craftcms/cms/discussions/11511#discussioncomment-3040404) for that.

**Comment by brandonkelly** [comment, marked as answer]
If you just want your Matrix content to be treated as search keywords for the entries, that’s possible out of the box.

Edit your Matrix field, and ensure it’s “Use this field’s values as search keywords” checkbox is ticked.

Then do the same for each sub-field that you want included in the Matrix field’s search keywords.

Finally, run the following command for any sections that use the Matrix field:

```sh
php craft resave/entries --section <sectionHandle> --update-search-index
```

Once finished, the `search` GraphQL query argument will work for your Matrix fields.

**Reply by DTesch-Level-R** [reply]
Thanks, that is neat 😀
Although I find myself happier using the Plugin <a href="https://plugins.craftcms.com/preparse-field" rel="nofollow">Preparse Field</a> because it gives me the opportunity to format the text / to strip html tags beforehand.

**Reply by brandonkelly** [reply]
OK, well in that case you can just mark the Preparse field as searchable.

**Reply by DTesch-Level-R** [reply]
Yes just checked, the html tags get stripped automatically. 😀

**Comment by bryantwells** [comment]
@brandonkelly would this technique work for entry fields nested within a matrix block?

**Reply by brandonkelly** [reply]
Sure, except the Preparse field isn’t going to automatically get updated when the related entries are updated.
